File: src/main/java/entities/Livraison1.java
package entities;

public class Livraison1 {
    private  int id_livraison; // Déclaré comme final car il est défini une fois dans le constructeur
    private  int id_commande;
    private  int id_client;
    private String date_livraison;
    private String adresse;
    private float prix;


    public Livraison1() {
    }

    public Livraison1(int id_livraison, int id_commande, int id_client, String date_livraison, String adresse, float prix) {
        this.id_livraison = id_livraison;
        this.id_commande = id_commande;
        this.id_client = id_client;
        this.date_livraison = date_livraison;
        this.adresse = adresse;
        this.prix = prix;
    }


    public Livraison1(int id_commande, int id_client, String date_livraison, String adresse, float prix) {
        this.id_commande = id_commande;
        this.id_client = id_client;
        this.date_livraison = date_livraison;
        this.adresse = adresse;
        this.prix = prix;
    }

    public int getId_livraison() {
        return id_livraison;
    }

    public void setId_livraison(int id_livraison) {
        this.id_livraison = id_livraison;
    }

    public int getId_commande() {
        return id_commande;
    }

    public void setId_commande(int id_commande) {
        this.id_commande = id_commande;
    }

    public int getId_client() {
        return id_client;
    }

    public void setId_client(int id_client) {
        this.id_client = id_client;
    }

    public String getDate_livraison() {
        return date_livraison;
    }

    public void setDate_livraison(String date_livraison) {
        this.date_livraison = date_livraison;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    @Override
    public String toString() {
        return "Livraison1{" +
                "id_livraison=" + id_livraison +
                ", id_commande=" + id_commande +
                ", id_client=" + id_client +
                ", date_livraison='" + date_livraison + '\'' +
                ", adresse='" + adresse + '\'' +
                ", prix=" + prix +
                '}';
    }
}

File: src/Entity/Livraison1.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Table(name: "livraison1")]
#[ORM\Entity(repositoryClass: "App\Repository\livraison1Repository")]
class Livraison1
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(name: "id_livraison", type: "integer", nullable: false)]
    private int $idLivraison;

    #[ORM\Column(name: "id_commande", type: "integer", nullable: false)]
    private int $idCommande;

    #[ORM\Column(name: "id_client", type: "integer", nullable: false)]
    private int $idClient;

    #[ORM\Column(name: "date_livraison", type: "string", length: 20, nullable: false)]
    private string $dateLivraison;

    #[ORM\Column(name: "adresse", type: "string", length: 20, nullable: false)]
    private string $adresse;

    #[ORM\Column(name: "prix", type: "float", precision: 10, scale: 0, nullable: false)]
    private float $prix;

    public function getIdLivraison(): ?int
    {
        return $this->idLivraison;
    }

    public function getIdCommande(): ?int
    {
        return $this->idCommande;
    }

    public function setIdCommande(int $idCommande): static
    {
        $this->idCommande = $idCommande;

        return $this;
    }

    public function getIdClient(): ?int
    {
        return $this->idClient;
    }

    public function setIdClient(int $idClient): static
    {
        $this->idClient = $idClient;

        return $this;
    }

    public function getDateLivraison(): ?string
    {
        return $this->dateLivraison;
    }

    public function setDateLivraison(string $dateLivraison): static
    {
        $this->dateLivraison = $dateLivraison;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): static
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;

        return $this;
    }
}
